src/block/inventory/FurnaceInventory.php

```diff
@@ -68,4 +68,26 @@ public function setFuel(Item $item) : void{
 	public function setSmelting(Item $item) : void{
 		$this->setItem(self::SLOT_INPUT, $item);
 	}
+
+	public function canAddSmelting(Item $item) : bool{
+		if($this->getSmelting()->isNull()) return true;
+
+		$currentInput = $this->getSmelting();
+
+		if($currentInput->getCount() === $currentInput->getMaxStackSize()) return false;
```

Comment:
This should not only check for equality but also if the count is greater than the max stack size.
```suggestion
		if($currentInput->getCount() >= $currentInput->getMaxStackSize()){
				return false;
		}
```